### PR TITLE
 add indices to jax.numpy frontend

### LINKED ISSUE #11872: indices

### PR DESCRIPTION
**Pull request by kunalagra**
Closes #11872

### CODE REVIEW

File: ivy/functional/frontends/jax/numpy/indexing.py [hunk 1]
```diff
@@ -30,3 +30,9 @@ def take_along_axis(arr, indices, axis, mode="fill"):
 @to_ivy_arrays_and_back
 def tril_indices(n_rows, n_cols=None, k=0):
     return ivy.tril_indices(n_rows, n_cols, k)
+
+
+# indices
+@to_ivy_arrays_and_back
+def indices(dimensions, dtype=<class 'jax.numpy.int32'>, sparse=False):
```
Comment (hmahmood24) [inline ivy/functional/frontends/jax/numpy/indexing.py:37]: The default value for `dtype` cannot be expressed this way. This will not be accepted by `pre-commit` when you submit your PR. It will throw a syntax error like it has [here](https://github.com/unifyai/ivy/actions/runs/4513332145/jobs/7948029470?pr=13091). The tests will also fail to be collected and `pytest` will also throw an error like it has [here](https://github.com/unifyai/ivy/actions/runs/4513523200/jobs/7948469962?pr=13091). Kindly change this to something like `jax.numpy.int32` and make sure that you [install pre-commit](https://github.com/unifyai/ivy/actions/runs/4513523200/jobs/7948469962?pr=13091) to ensure that these formatting checks get carried out automatically before you push your code and also run your tests locally first to ensure that they are being collected and are passing for all backends before submitting a PR. 🙂

File: ivy/functional/frontends/jax/numpy/indexing.py [hunk 2]
```diff
@@ -30,3 +30,9 @@ def take_along_axis(arr, indices, axis, mode="fill"):
...
+    return ivy.indices(dimensions, dtype, sparse)
```
Comment (hmahmood24) [inline ivy/functional/frontends/jax/numpy/indexing.py:38]: Are you sure you ran the tests locally and verified them? `ivy` does not have any methods named `indices`. If you run the tests, it will throw you this error right away. In this scenario, you will have to come up with a manual implementation of `indices` using just `ivy` methods. You can look into `jax`'s source code for reference [here](https://jax.readthedocs.io/en/latest/_modules/jax/_src/numpy/lax_numpy.html#indices) or go through Ivy's [docs](https://lets-unify.ai/ivy/) to look for the closest function in any other frontends or the functional API to what you're trying to implement. 🙂

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @kunalagra! Thanks for submitting a PR! I have requested some changes for you to look into. Kindly review and re-request a review from me once you've done implementing them and ran the linting checks as well as the tests for your function locally. Good luck! 😊

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
